### DISCUSSION TITLE
Pico do not start if sun voltage rises slowly

### DISCUSSION THREAD

**Discussion by whallmann** [EngineerGuy314/pico-WSPRer#36] [General]
With the voltage divider resistors i watched with a new bunch of original picos that they do not start at sun rise.  I played around with resistor values but no idea.  ecerything is fine if i hold the pico fast into sun light. but if i would do this very very slowly it do nothing.
Tried with two different picos.  Someone else see this effect? What can i do?
cu Wolf

**Comment by EngineerGuy314** [comment]
how many pins are on the voltage regulator chip? 6 or 10? I have a few "genuine" picos that have the same 6 pin chip as the clones, and they will not work.

Also, how are you enabling the GPS module? MOSFET or 3 IO pins in parallel? If you take a shortcut and skip that step it will behave how you describe.

![PMIC_chips](https://github.com/user-attachments/assets/0c0289fb-392c-4e4a-89f4-797573a5b66b)

**Comment by whallmann** [comment]
It looks as we have here a good one. 10 Pins are visible.
I am always using the mosfet N7000 for GPS.
Just found the first pcb i started last year experiments. compared to the
one who made this problems, i cant see any difference.  added two Pictures.

First one is the old good working pico.






EngineerGuy314 ***@***.***> schrieb am Do., 29. Aug. 2024,
22:02:

**Reply by whallmann** [reply]
Here are the missing pictures:
First: this is one of my first experiment pcbs. 
![image](https://github.com/user-attachments/assets/6435669f-8ca0-46bc-bef1-7613b1595d64)
Second:  this is one of the newest delivery from same company 
![image](https://github.com/user-attachments/assets/4c2aadb4-1bed-4335-b49e-c77a0c9481ca)

**Comment by whallmann** [comment]
Bad news on this theme. I invested much hours to get more details. The pico powerup process does not start up every time . It hangs up sometimes. No RUN pin selection against ground helps or reanimates the startup process again. So in my opinion the last production of original raspi picos is not longer usable for this project.

tldr;
What i have done:
I disconnected the voltage divider resistors.
Gave 4 Volts on vsys port (2.nd from top side) . Then shortened the RUN against Ground. The 3 seconds init blink started, one, twice ... then some more... it stops. Checked the voltage on vsys who is still normal. the 3v3 pin for gps is available. So the init should start the main program . but no chance on use the RUN connect to ground. 
A next try is only posible after disconnect the vsys for a moment.
The picos where bought from different companies they saying this are geniue once.
The china modells are good clones but you can detect they are clones. So i am sure thus are good ones.
What i will say, it looks like the specifications are changed anyway.  
Some more tests with voltage supervisor ic's like max809 are a solution. On a slow voltage rising while the pico startup sequence runce more then once, the max809 pushes the reset switch RUN pin to ground if the defined voltage e.g. 3 or 4 volt is reached. The RUN push takes no effect anymore.  I also tried combinations of the voltage divider and max309 chip, on slow voltage rising it hangs up. On fast rising everything works. 

I gave up at this moment. What i need are some fresh ideas.

Sorry for long text. I checked so much combinations, i cant tell about all.

**Reply by EngineerGuy314** [reply]
Sorry to hear about your problems, I know how frustrating that kind of thing can be.

Your previous post mentioned you attached two pictures, but they didn't come through, so I can't see how you have the test wired up. I only mention that in case you are doing this test on a breadboard or with long wires (instead of two resistors soldered directly to the pico).  We had seen some strange things where the resistors connected to the RUN input were picking up interference and causing erratic behavior.  I had one test setup that was working, except as soon as I touched the high impedance probe of my oscilloscope to the run input (with the resistors connected) the pico would reboot. it was that sensitive.

I always said that I never could get fake picos to work, but all of the real ones I used (about 8) worked well. But that doesn't prove anything. Like you said, there is no way of knowing if even the real picos are slightly different in the newer batches. I used 8 real picos and they all worked, but thats not the same as saying I used 8,000 picos and they all worked!

The RP2040, like many other microcontrollers, do not like gradual voltage rise. I completely admit that using two resistors to prevent lockup in this case is a very crude method and not very robust. It is probably just on the border of being able to work at all. I was actually very surprised when i launched the first one and it actually woke up. 

Using a max809 supervisor would be much more reliable than the resistors. Thats what I have on my custom PCB version of the pico-WSPRer, and it works great. If you used one on your setup I am sure it would be a rock solid solution.

Here is how I have it hooked up on the PCB version. It requires an external capacitor and resistor.
![max809](https://github.com/user-attachments/assets/4e63729a-55eb-4912-9a2e-63673f3913e8)

I never wanted to add this to the design of the "original" pico-WSPRer, because it seemed that it was getting to the point of having to buy and solder so many separate components it was worth it to just make a PCB. But in your case it might be necessary.

**Reply by whallmann** [reply]
I know that short cables are essential. But after 5 picos in the air, all build with same wireing, they should do the same i expect.

>" _I completely admit that using two resistors to prevent lockup in this case is a very crude method and not very robust._ "
 Ok, but it works and as i investigated the values between to much or to low voltage at the RUN pin was good enough.  

>"_Using a max809 supervisor would be much more reliable than the resistors. Thats what I have on my custom PCB version of the pico-WSPRer, and it works great. If you used one on your setup I am sure it would be a rock solid solution._ "
yes, agree. But if the start up process hangs, it doesnt help if the RUN pin shows no reaction anymore.  After a hang, i messured the voltages. Every value was above the limits but the startup process runs like in a loop:   blinking 3 seconds, pause 1 second, blinking 3 seconds and so on, looks like he did not start the real programm loaded up.  
I worked all the time with 6 x solar panels 0,5 volt = 3 Volts, but in the sun it came up to 3,7 Volts.  At my last tests i added a plate more showing in sun up to 4,5 Volts. This was nessesary because the max809 delivers only 3,1 Volt and 4,1 Volt tresholds for RESET. I used the MAX809FLEUR Version - with inverted RESET. Pulling RUN Pin to Ground for 200ms if T-Value is reached.  No more Resistors or capacity like your solution. Maybe i trapped here in my thoughts :-)

PIC: 
![image](https://github.com/user-attachments/assets/b56c208e-cce3-4803-bf0e-253aadedf205)
One of my wireings 

My question in priority is why the start up process is repeating more then once until the software is starting. With a 4 Volt treshold max809 connected i see something strange:  the value 4,1 Volt for the max809 is reached, the max is pushing the RESET (and so RUN to GROUND) - the voltage on the VSYS is dropping short below 4 Volts, then up again, the max treshold reached again, the RESET is pushed again, and so on.  Maybe this is the problem for starting again and again? You have any idea to check this on your side? What can i do against the voltage drop?

If i hold the panels some more into light, the volts value from the panels increases a little more, the voltage drop on RESET stays ober 4 Volts and then the software is starting. This question torments me all the time.

Need some pause, and will invest more time later... tnx

**Reply by EngineerGuy314** [reply]
>>My question in priority is why the start up process is repeating more then once until the software is starting

short answer: the pico boots, attempts to enable the GPS after a couple seconds, and if the sun isn't strong enough to provide enough mA for both the pico and GPS, it reboots and tries again. This is how it "checks" to see if there is enough sun to provide full current. Eventually the sun will be strong enough to allow both to be powered on simultaneously. but we can't reliably "know" when the sun is strong enough to produce enough current without trying. So it will repeat the cycle.

Long Answer: 
Without the MOSFET, the GPS module would start to draw current from the regulator even before the RP2040 has started booting.  This immediate and large current draw would make the voltage rise from the cells very slowly, and put the RP2040 into the locked condition (and once its locked, even if the voltage continues to rise up to 5V, it will stay locked).

So the MOSFET lets the pico keep the GPS module un-powered until the pico has booted up and spent a couple seconds blinking its LED.

But every morning during sunrise there will be the situation that the cells are able to produce enough voltage and current to power the pico, but not enough to power both the pico and GPS. So after a couple seconds the pico enables the MOSFET, but its sudden current draw makes the voltage drop below the threshold for the resistor divider to enable the RUN input, so the pico reboots. When it reboots, it also kills the output to the MOSFET, so the GPS stops drawing current and the voltage shoots up again. 

So the cycle repeats itself. And for the entire time that the sun is up (but still weak) there is an endless cycle of the pico booting, doing some blinky, and then rebooting when it tries (momentarily) to 

Why does this system work? My theory:  even though the voltage output of the cells rises and drops during these cycles, it rises and drops rapidly enough to prevent the RP2040 from entering the dreaded locked state.

How well does this system work? It probably barely works at all, and when it does work we are just lucky. A different pico from a different batch unfortunately might not work.

**Comment by EngineerGuy314** [comment]
>> I used the MAX809FLEUR Version - with inverted RESET. Pulling RUN Pin to Ground for 200ms if T-Value is reached

I couldn't find info on a MAX809FLEUR, but if thats really how it functions (inverted) I don't think that will work.

I admit I don't know much about the function of the max809. I copied large parts of my PCB schematic, including the max809 circuit, from someone else's design. It just worked, so I never thought much about it.

**Comment by whallmann** [comment]
Some news about the "not starting Pico models".

I tried the last days with a *max809teur* and also *SGM809B-TXN3LG* voltage
supervisor. After some disappointments in the morning, my thoughts were that
the voltage drops too hard down if the  supervisor  switches the RUN Button
to GROUND. It looks like the voltage dropdown lets it restart itself.
So I played around with some resistor values between the RESET Pin of the
supervisor  and RUN Pin. I started with 10 kOhm. Yesterday I added some
more. Meanwhile I am using 40 kOhms. Now I have my first self start at
sunrise.  Hope to see some more in the next few days.  I am hopefully :-)
Maybe this also helps to get the clone Picos running, but one after each
other.

**Reply by whallmann** [reply]
_Correcting_:  the SGM was an other type:  SGM809B-JXN3LG (4 Volt Threshold) - because i used 8 solar panels
